### DISCUSSION TITLE
Recursive Pointer

### DISCUSSION THREAD

**Discussion by jeffreyjackson** [netreconlab/Parse-Swift#127] [General]
Hello!  First off, thank you @cbaker6 for your efforts on this SDK.  Ive pulled it in for a new front-end project to try it out.  This project is using an existing backend thats been around for a few years. 

I understand my question goes against the grain regarding value-types for swift, but coming from a previous design on the old school SDK - im trying to figure out the correct approach to this since the backend architecture and object names are already in place.  

The example I'm trying to address is similar to a recursive representation of departments.   

Simply:
```
struct Organization: ParseObject {
    init() {
        
    }
    // omitting default fields here ...
    
    // custom fields 
    var organizationDescription: String
    var parentOrganization: Organization?  <-- this is implemented and works as intended with the original SDK
}
```

I understand I could includeKey in the query previously, which isn't necessary for this new frontEnd Im working on.  However, the user could potentially query for organization where they all match a specific parentOrganization of the same type.  Alternatively for a given organization, I could query for the parent if needed.  

What would be your recommendation to overcome this current design pattern?

**Comment by cbaker6** [comment]
Sure, I’ve answered similar questions in the past, below are some of the answers:

- https://github.com/netreconlab/Parse-Swift/discussions/109#discussioncomment-5934797
- https://github.com/parse-community/Parse-Swift/issues/161#issuecomment-859653209
- https://community.parseplatform.org/t/pulling-data-out-of-pointer-object-without-having-to-fetch-again/2191

**Reply by jeffreyjackson** [reply]
thank you @cbaker6 !
